File: testresources/csharp/benchmark/Domain/Direct/Violating/AccessLocalVariable_SetArgumentValue.cs
namespace Domain.Direct.Violating
{
	using Technology.Direct.Dao;	

	public class AccessLocalVariable_SetArgumentValue
	{

	public string theMethod(string s, int i) {
		ProfileDAO profile;
		string p1 = getProfileInformation(profile, "test", 4).checkinValue;
		return p1;
	}
	
	public string getProfileInformation(string s, ProfileDAO profile, int i) {
		string p2 = s;
		return p2;
	}

	public ProfileDAO getProfileInformation(ProfileDAO profile, string s, int i) {
		string p3 = s;
		return p3;
	}
	}
}

File: src/test/resources/java/accuracy/src/domain/direct/violating/AccessLocalVariable_SetArgumentValue.java
package domain.direct.violating;

import technology.direct.dao.ProfileDAO;

public class AccessLocalVariable_SetArgumentValue {
	
	// Tests if: 1) the type of argument profile is found; 2) thereafter the right method is selected; 3) the return value of this method is found.
	public String theMethod(String s, int i) {
		ProfileDAO profile = null;
		// Access profile is direct, but of checkinvalue is indirect.
		String pn = getProfileInformation(profile, "test", 4).checkinValue.name;
		return pn;
	}
	
	public String getProfileInformation(String s, ProfileDAO profile, int i) {
		String pn = s;
		return pn;
	}

	public ProfileDAO getProfileInformation(ProfileDAO profile, String s, int i) {
		return profile;
	}
}